Src/IronPython.Modules/nt.cs

```diff
@@ -1317,36 +1475,45 @@ public sealed class terminal_size : PythonTuple {
             throw GetLastUnixError(path);
         }
 
-        public static void utime(string path, [ParamDictionary]IDictionary<string, object> dict)
-            => utime(path, null, dict);
+        [Documentation("utime(path, times=None, *[, ns], dir_fd=None, follow_symlinks=True)")]
+        public static void utime([NotNull] string path, [ParamDictionary] IDictionary<string, object> kwargs, params object[] args) {
+            var numArgs = args.Length;
+            CheckOptionalArgsCount(numRegParms: 1, numOptPosParms: 1, numKwParms: 3, numArgs, kwargs.Count);
+
+            PythonTuple times = numArgs == 0 || args[0] == null ? null
+                                : args[0] is PythonTuple pt1 ? pt1
+                                : throw PythonOps.TypeError("utime: 'times' must be either a 2-value tuple (atime, mtime) or None");
 
-        public static void utime(string path, PythonTuple times, [ParamDictionary]IDictionary<string, object> dict) {
             PythonTuple ns = null;
-            if (dict != null) {
-                foreach (var pair in dict) {
-                    switch (pair.Key) {
-                        case "ns":
-                            if (times != null) {
-                                throw PythonOps.ValueError("utime: you may specify either 'times' or 'ns' but not both");
-                            }
-                            if (pair.Value is PythonTuple t && t.__len__() == 2) {
-                                ns = t;
-                            } else {
-                                throw PythonOps.TypeError("utime: 'ns' must be a tuple of two ints");
-                            }
-                            break;
-                        case "dir_fd":
-                        case "follow_symlinks":
-                            // TODO: implement these!
-                            break;
-                        default:
-                            throw PythonOps.TypeError("'{0}' is an invalid keyword argument for this function", pair.Key);
-                    }
+            foreach (var kvp in kwargs) {
+                switch (kvp.Key) {
+                    case nameof(times):
+                        if (numArgs > 0) throw PythonOps.TypeError("argument for {0}() given by name ('{1}') and position ({2})", nameof(utime), nameof(times), 2);
+                        times = kvp.Value == null ? null
+                                : args[0] is PythonTuple pt2 ? pt2
+                                : throw PythonOps.TypeError("utime: 'times' must be either a 2-value tuple (atime, mtime) or None");
```

Comment:
Should `args[0]` here be `kvp.Value`? Maybe wrap the logic in a local function:
```C#
static PythonTuple GetTimes(object val) {
    return val == null ? null
        : val is PythonTuple pt ? pt
        : throw PythonOps.TypeError("utime: 'times' must be either a 2-value tuple (atime, mtime) or None");
}
```
I guess we could also do the length check on the tuple in there?

Reply:
OK. In general, I am still working on the keyword/default parameter handling and am musing about some generic, multipurpose mechanism that will cut though so much ceremony that is now required. The `\u00F8` seems not to work anymore since Python 3 allows Unicode letters in identifiers.  But perhaps I will start with something simpler, like #882.